src/java.desktop/share/classes/javax/swing/text/html/CSS.java

```diff
@@ -2678,23 +2678,16 @@ Object toStyleConstants(StyleConstants key, View v) {
 
         @Override
         public int hashCode() {
-            int hashCode = Float.hashCode(span) | Boolean.hashCode(percentage);
-            if (units != null) {
-                hashCode |= units.hashCode();
-            }
-            return hashCode;
+            return Float.hashCode(span)
+                   | Boolean.hashCode(percentage)
+                   | Objects.hashCode(units);
         }
 
         @Override
         public boolean equals(Object val) {
-            if (percentage) {
-                return val instanceof CSS.LengthValue lu
-                       && Objects.equals(svalue, lu.svalue);
-            } else {
-                return val instanceof CSS.LengthValue lu
-                       && span == lu.span
-                       && Objects.equals(units, lu.units);
-            }
+            return val instanceof CSS.LengthValue lu
+                   && span == lu.span
+                   && Objects.equals(units, lu.units);
```

Comment:
[The `percentage` field](https://github.com/openjdk/jdk/pull/13405#discussion_r1221932524) must also be part of `equals`:
```suggestion
            return val instanceof CSS.LengthValue lu
                   && percentage == lu.percentage
                   && span == lu.span
                   && Objects.equals(units, lu.units);
```
You have included it in `hashCode`.

Reply:
If `percentage` isn't taken into account, the pair `{"margin-top: 100%", "margin-top: 1"}` is considered equal.

Reply:
Updated..